packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -0,0 +1,188 @@
+import React, {useState, useRef, useEffect} from 'react'
+import clsx from 'clsx'
+import type {BaseProps} from '../component-helpers'
+import {Text} from '../Text'
+import {Controls} from './components'
+import {MarkGithubIcon} from '@primer/octicons-react'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/video-player/base.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './VideoPlayer.module.css'
+
+type VideoPlayerProps = {
+  poster?: string
+  title: string
+  branding?: boolean
+  children: React.ReactElement | React.ReactElement[]
+  ref?: React.RefObject<HTMLVideoElement>
+} & React.HTMLProps<HTMLVideoElement> &
+  BaseProps<HTMLVideoElement>
+
+const Root = ({
+  poster,
+  title,
+  branding = true,
+  children,
+  className,
+  onPlay,
+  onPause,
+  onLoadedMetadata,
+  onPlaying,
+  ref,
+  ...rest
+}: VideoPlayerProps) => {
+  const videoPlayerRef = useRef<HTMLVideoElement>(null)
+  const videoWrapperRef = useRef<HTMLDivElement>(null)
+  const videoRef = ref ? ref : videoPlayerRef
+  const [playing, setPlaying] = useState(false)
+  const [playedTime, setPlayedTime] = useState(0)
+  const [totalTime, setTotalTime] = useState(0)
+  const [trackInformation, setTrackInformation] = useState<TextTrackCueList | undefined>(undefined)
+
+  /* > Play / Pause                                                    */
+  const handleVideoPlayback = () => {
+    if (videoRef.current) {
+      if (videoRef.current.paused) {
+        videoRef.current.play()
+      } else {
+        videoRef.current.pause()
+      }
+    }
+  }
+
+  const handleVideoOnPlaying = () => {
+    videoRef.current?.textTracks[0].cues && setTrackInformation(videoRef.current.textTracks[0].cues)
+  }
+
+  /* > Hide Default Captions                                                   */
+  useEffect(() => {
+    if (videoRef.current) {
+      for (let i = 0; i < videoRef.current.textTracks.length; i++) {
+        videoRef.current.textTracks[0].mode = 'hidden'
+      }
+    }
+  }, [videoRef])
```

Comment:
I'm seeing caption-related DOM update events when captions are toggled to off. I imagine this is unintended?

https://github.com/primer/brand/assets/13340707/04077d41-dcba-492a-a350-ece446b240da

Reply:
This is actually intended. So to get access to them completely as well as the ability for them to interact with our other controls they are a DOM element.

Reply:
Sorry, to clarify I'm referring to the the caption DOM elements being updated when **captions are off.** That doesn't feel intended, as the user has turned them off and it's creating additional, and unnecessary computation. Could you add a guard check to see if captions are enabled/disabled before running that effect?

Reply:
I can make this change the tradeoff would be that if a user turns on captions when the update is not running then they won't have access to the captions at that moment. Is this a tradeoff that we want to accept? I would think that more immediate captions would be preferred from an accessibility standpoint. @ericwbailey do you know what is preferred in this case?

Reply:
From an end user perspective we'll want to ensure that the end effect of invoking captions leads to them appearing immediately on request and synchronized to the currently spoken audio.

I'm unsure about the performance implication particulars, or if a WebVTT/caption file is smart enough to jump immediately to the timestamp. I don't want to crash the browser or AT, either.

Reply:
I don't believe this has any heavy browser implications but it does supply the captions immediately when the other approach would wait for the next captions to be sent once they are turned on. So I think the current approach is preferred from a11y but has the tradeoff of slightly negative performance. @rezrah I think this is an ok tradeoff, do you have any major concerns?

Reply:
> when the update is not running then they won't have access to the captions at that moment.

Do you know what the exact impact of this is, specifically how long the delay for the captions to appear, catch up and synchronise is? I'm a little curious if and why the delay would even be noticeable, but if it is not within a second (for example), **happy to leave as-is**. A major concern with any kind of side effect in React is susceptibility to memory leaks. Thinking defensively, i'd prefer that we don't mutate dom elements directly unless we absolutely have to, but happy to leave as-is if we can't achieve synchronisation as Eric alluded to.